src/util.ts

```diff
@@ -0,0 +1,62 @@
+import BN = require("bn.js");
+
+// Geth compatible DB keys
+
+const headsKey = "heads";
+/**
+ * Current canonical head for light sync
+ */
+const headHeaderKey = "LastHeader";
+/**
+ * Current canonical head for full sync
+ */
+const headBlockKey = "LastBlock";
+/**
+ * headerPrefix + number + hash -> header
+ */
+const headerPrefix = Buffer.from("h");
+/**
+ * headerPrefix + number + hash + tdSuffix -> td
+ */
+const tdSuffix = Buffer.from("t");
+/**
+ * headerPrefix + number + numSuffix -> hash
+ */
+const numSuffix = Buffer.from("n");
+/**
+ * blockHashPrefix + hash -> number
+ */
+const blockHashPrefix = Buffer.from("H");
+/**
+ * bodyPrefix + number + hash -> block body
+ */
+const bodyPrefix = Buffer.from("b");
+
+// Utility functions
+
+/**
+ * Convert BN to big endian Buffer
+ */
+const bufBE8: (n: BN) => Buffer = (n: BN) => n.toArrayLike(Buffer, "be", 8);
+const tdKey = (n: BN, hash: Buffer) =>
+  Buffer.concat([headerPrefix, bufBE8(n), hash, tdSuffix]);
+const headerKey = (n: BN, hash: Buffer) =>
+  Buffer.concat([headerPrefix, bufBE8(n), hash]);
+const bodyKey = (n: BN, hash: Buffer) =>
+  Buffer.concat([bodyPrefix, bufBE8(n), hash]);
+const numberToHashKey = (n: BN) =>
+  Buffer.concat([headerPrefix, bufBE8(n), numSuffix]);
+const hashToNumberKey = (hash: Buffer) =>
+  Buffer.concat([blockHashPrefix, hash]);
+
+export {
+  headsKey,
+  headHeaderKey,
+  headBlockKey,
+  bufBE8,
+  tdKey,
+  headerKey,
+  bodyKey,
+  numberToHashKey,
+  hashToNumberKey
+};
```

Comment:
Why is there a type annotation only for ``bufBE8``? I generally find this ``bufBE8`` line hard to read for a non-TypeScript pro.

Reply:
I don't understand, which line are you referring to?

Reply:
``const bufBE8: (n: BN) => Buffer = (n: BN) => n.toArrayLike(Buffer, "be", 8);`` (the ``: (n: BN) => Buffer`` is a type annotation if I am not mistaken?) vs ``const tdKey = (n: BN, hash: Buffer) =>
  Buffer.concat([headerPrefix, bufBE8(n), hash, tdSuffix]);`` and the rest.

Reply:
I still don't understand the question. What part of the type annotation is weird?

Reply:
Updated to make it a bit less verbose